modules/darwinosl/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-if(APPLE)
+if(ENABLE_OBJC)
```

Comment:
I think this must be

if (ENABLE_OBJC AND ENABLE_DARWIN_OSL)

Reply:
But they are not independent. `ENABLE_DARWIN_OSL` cannot work without ObjC, and `ENABLE_OBJC` is set by `ENABLE_DARWIN_OSL` value. Do I miss something?

Reply:
yes, it s the case now, but with these modifications, `ENABLE_OBJC` can depend on multiple future (ObjC) modules

that is why i suggested using a separate `ENABLE_OBJC` variable here

Reply:
Ok, got it. I will deal with this today then.